api/app/weather_models/process_grib.py

```diff
@@ -148,6 +163,53 @@ def yield_data_for_stations(self, raster_band: gdal.Dataset):
 
             yield (points, values)
 
+    def calculate_wind_speed_from_u_v(self, u: float, v: float):
+        """ Return calculated wind speed from u and v components using formula
+        wind_speed = sqrt(u^2 + v^2)
+        """
+        return math.sqrt(math.pow(u, 2) + math.pow(v, 2))
+
+    def calculate_wind_dir_from_u_v(self, u: float, v: float):
+        """ Return calculated wind direction from u and v components using formula
+        wind_direction = arctan(u, v) * 180/pi (in degrees)
+        """
+        calc = math.atan2(v, u) * 180 / math.pi
+        return calc if calc > 0 else calc + 360
+
+    def yield_uv_wind_data_for_stations(self, u_raster_band: gdal.Dataset, v_raster_band: gdal.Dataset, variable: str):
+        """ Given a list of stations and 2 gdal datasets (one for u-component of wind, one for v-component
+        of wind), yield relevant data 
+
+        What the heck is going on here?! See
+        http://colaweb.gmu.edu/dev/clim301/lectures/wind/wind-uv
+        """
+        for station in self.stations:
+            longitude = station.long
+            latitude = station.lat
+
+            x_coordinate, y_coordinate = calculate_raster_coordinate(
+                longitude, latitude, self.padf_transform, self.geo_to_raster_transformer)
+
+            if 0 <= x_coordinate < u_raster_band.XSize and 0 <= x_coordinate < v_raster_band.XSize and\
+                    0 <= y_coordinate < u_raster_band.YSize and 0 <= y_coordinate < v_raster_band.YSize:
+                u_points, u_values = get_surrounding_grid(u_raster_band, x_coordinate, y_coordinate)
+                v_points, v_values = get_surrounding_grid(v_raster_band, x_coordinate, y_coordinate)
+
+                if variable == 'wdir_tgl_10':
+                    wind_dir_values = []
+                    for u, v in zip(u_values, v_values):
+                        wind_dir_values.append(self.calculate_wind_dir_from_u_v(u, v))
+                    if u_points == v_points:
+                        yield (u_points, wind_dir_values)
+                elif variable == 'wind_tgl_10':
+                    wind_sp_values = []
```

Comment:
Does `sp` stand for `speed`?

Reply:
it does, but I'll rename that for clarity